Sources/TuistDependencies/Carthage/CarthageInteractor.swift

```diff
@@ -0,0 +1,195 @@
+import RxBlocking
+import TSCBasic
+import TuistCore
+import TuistSupport
+
+// MARK: - Carthage Interactor Errors
+
+enum CarthageInteractorError: FatalError, Equatable {
+    /// Thrown when Carthage cannot be found.
+    case carthageNotFound
+    /// Thrown when Carfile cannont be found in temporary directory after Carthage installation.
+    case cartfileNotFound
+    /// Thrown when `Carthage/Build` directory cannont be found in temporary directory after Carthage installation.
+    case buildDirectoryNotFound
+
+    /// Error type.
+    var type: ErrorType {
+        switch self {
+        case .carthageNotFound, .cartfileNotFound, .buildDirectoryNotFound:
+            return .abort
+        }
+    }
+
+    /// Error description.
+    var description: String {
+        switch self {
+        case .carthageNotFound:
+            return "Carthage was not found in the environment. It's possible that the tool is not installed or hasn't been exposed to your environment."
+        case .cartfileNotFound:
+            return "Cartfile was not found after Carthage installation."
+        case .buildDirectoryNotFound:
+            return "Carthage/Build directory was not found after Carthage installation."
+        }
+    }
+}
+
+// MARK: - Carthage Interacting
+
+public protocol CarthageInteracting {
+    /// Installes `Carthage` dependencies.
+    /// - Parameter dependenciesDirectory: The path to the directory that contains the `Tuist/Dependencies/` directory.
+    /// - Parameter method: Installation method.
+    /// - Parameter dependencies: List of dependencies to intall using `Carthage`.
+    func install(dependenciesDirectory: AbsolutePath, method: InstallDependenciesMethod, dependencies: [CarthageDependency]) throws
+}
+
+// MARK: - Carthage Interactor
+
+public final class CarthageInteractor: CarthageInteracting {
+    private let fileHandler: FileHandling
+    private let carthageCommandGenerator: CarthageCommandGenerating
+    private let cartfileContentGenerator: CartfileContentGenerating
+
+    public init(
+        fileHandler: FileHandling = FileHandler.shared,
+        carthageCommandGenerator: CarthageCommandGenerating = CarthageCommandGenerator(),
+        cartfileContentGenerator: CartfileContentGenerating = CartfileContentGenerator()
+    ) {
+        self.fileHandler = fileHandler
+        self.carthageCommandGenerator = carthageCommandGenerator
+        self.cartfileContentGenerator = cartfileContentGenerator
+    }
+
+    public func install(dependenciesDirectory: AbsolutePath, method: InstallDependenciesMethod, dependencies: [CarthageDependency]) throws {
+        switch method {
+        case .fetch: logger.info("Start fetching Carthage dependencies.", metadata: .section)
```

Comment:
Similar comment in regards to the tense of the logs. You are telling the user something; if we use infinitive, the user doesn't feel like we are telling them something.